flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/runtime/batch/sql/PartitionableSinkITCase.scala

```diff
@@ -0,0 +1,278 @@
+      .setInteger(ExecutionConfigOptions.SQL_RESOURCE_DEFAULT_PARALLELISM, 3)
+    registerCollection("nonSortTable", testData, type3, "a, b, c", dataNullables)
+    registerCollection("sortTable", testData1, type3, "a, b, c", dataNullables)
+    PartitionableSinkITCase.init()
+  }
+
+  override def getTableConfig: TableConfig = {
+    val parserConfig = SqlParser.configBuilder
+      .setParserFactory(FlinkSqlParserImpl.FACTORY)
+      .setConformance(FlinkSqlConformance.HIVE) // set up hive dialect
+      .setLex(Lex.JAVA)
+      .setIdentifierMaxLength(256).build
+    val plannerConfig = CalciteConfig.createBuilder(CalciteConfig.DEFAULT)
+      .replaceSqlParserConfig(parserConfig)
+    val tableConfig = new TableConfig
+    tableConfig.setPlannerConfig(plannerConfig.build())
+    tableConfig
+  }
+
+  @Test
+  def testInsertWithOutPartitionGrouping(): Unit = {
+    registerTableSink(grouping = false)
+    tEnv.sqlUpdate("insert into sinkTable select a, max(b), c"
+      + " from nonSortTable group by a, c")
+    tEnv.execute("testJob")
+    val resultSet = List(RESULT1, RESULT2, RESULT3)
+    assert(resultSet.exists(l => l.size() == 3))
+    resultSet.filter(l => l.size() == 3).foreach{ list =>
+      assert(list.forall(r => r.getField(0).toString == "1"))
+    }
+  }
+
+  @Test
+  def testInsertWithPartitionGrouping(): Unit = {
+    registerTableSink(grouping = true)
+    tEnv.sqlUpdate("insert into sinkTable select a, b, c from sortTable")
+    tEnv.execute("testJob")
+    val resultSet = List(RESULT1, RESULT2, RESULT3)
+    resultSet.foreach(l => assertSortedByFirstNField(l, 1))
+    assertEquals(resultSet.map(l => collectDistinctGroupCount(l, 2)).sum, 4)
+  }
+
+  @Test
+  def testInsertWithStaticPartitions(): Unit = {
+    val testSink = registerTableSink(grouping = true)
+    tEnv.sqlUpdate("insert into sinkTable partition(a=1) select b, c from sortTable")
+    tEnv.execute("testJob")
+    // this sink should have been set up with static partitions
+    assertEquals(testSink.getStaticPartitions.toMap, Map("a" -> "1"))
+    val resultSet = List(RESULT1, RESULT2, RESULT3)
+    val result = resultSet.filter(l => l.size() == 11)
+    assert(result.size == 1)
+    result.get(0).forall(r => r.getField(0).asInstanceOf[Int] == 1)
+  }
+
+  @Test
+  def testInsertWithStaticAndDynamicPartitions(): Unit = {
+    val testSink = registerTableSink(grouping = true, partitionColumns = Array("a", "b"))
+    tEnv.sqlUpdate("insert into sinkTable partition(a=1) select b, c from sortTable")
+    tEnv.execute("testJob")
+    // this sink should have been set up with static partitions
+    assertEquals(testSink.getStaticPartitions.toMap, Map("a" -> "1"))
+    val resultSet = List(RESULT1, RESULT2, RESULT3)
+    resultSet.foreach(l => assertSortedByFirstNField(l, 2))
+    assertEquals(resultSet.map(l => collectDistinctGroupCount(l, 2)).sum, 4)
+  }
+
+  @Test(expected = classOf[TableException])
+  def testDynamicPartitionInFrontOfStaticPartition(): Unit = {
```

Comment:
please assert the exception message as well.

Reply:
Sorry i think this is overchecked

Reply:
Added